FinSearchUnified/Bundle/SearchBundleFindologic/QueryBuilder/NewQueryBuilder.php

```diff
@@ -0,0 +1,240 @@
+        Client $apiClient = null
+    ) {
+        $this->installerService = $installerService;
+        $this->config = $config;
+        $this->shopKey = $config->offsetGet('ShopKey');
+        $this->shopUrl = rtrim(Shopware()->Shop()->getHost(), '/');
+
+        if ($apiClient === null) {
+            $apiConfig = new Config();
+            $apiClient = new Client($apiConfig);
+        }
+
+        $this->apiClient = $apiClient;
+        $this->setDefaultParameters();
+    }
+
+    /**
+     * @return Response
+     */
+    public function execute()
+    {
+        return $this->apiClient->send($this->searchNavigationRequest);
+    }
+
+    /**
+     * @param array $categories
+     */
+    abstract public function addCategories(array $categories);
+
+    /**
+     * @return array
+     */
+    public function getParameters()
+    {
+        return $this->searchNavigationRequest->getParams();
+    }
+
+    /**
+     * @param string $query
+     */
+    public function addQuery($query)
+    {
+        $this->searchNavigationRequest->setQuery($query);
+    }
+
+    /**
+     * @param string $key
+     * @param float $min
+     * @param float $max
+     */
+    public function addRangeFilter($key, $min, $max)
+    {
+        $this->searchNavigationRequest->addAttribute($key, $min, 'min');
+        $this->searchNavigationRequest->addAttribute($key, $max, 'max');
+    }
+
+    /**
+     * @param string $key
+     * @param mixed $values
+     */
+    public function addFilter($key, $values)
+    {
+        if (!is_array($values)) {
+            $values = [$values];
+        }
+
+        foreach ($values as $value) {
+            $this->addParameter($key, urldecode($value));
+        }
+    }
+
+    /**
+     * @param string $order
+     */
+    public function addOrder($order)
+    {
+        $this->searchNavigationRequest->setOrder($order);
+    }
+
+    /**
+     * @param int $firstResult
+     */
+    public function setFirstResult($firstResult)
+    {
+        $this->searchNavigationRequest->setFirst($firstResult);
+    }
+
+    /**
+     * @param int $maxResults
+     */
+    public function setMaxResults($maxResults)
+    {
+        $this->searchNavigationRequest->setCount($maxResults);
+    }
+
+    /**
+     * @param string $key
+     *
+     * @return mixed|null
+     */
+    public function getParameter($key)
+    {
+        $params = $this->searchNavigationRequest->getParams();
+
+        return $params[QueryParameter::ATTRIB][$key];
+    }
+
+    /**
+     * @param string $key
+     * @param mixed $value
+     */
+    public function addParameter($key, $value)
+    {
+        $this->searchNavigationRequest->addAttribute($key, $value);
+    }
+
+    /**
+     * @param string $usergroup
+     */
+    public function addUserGroup($usergroup)
+    {
+        $hashedGroup = StaticHelper::calculateUsergroupHash($this->shopKey, $usergroup);
+        $this->searchNavigationRequest->addGroup($hashedGroup);
```

Comment:
This one should actually be `addUserGroup`. I think FINDOLOGIC-API does not yet support this parameter, so please add another PR that adds this parameter. Thanks.

Reply:
The name of the parameter is `usergrouphash` and the method should be `addUserGroup`.